tests/testsuite.src/syn_misc.at

```diff
@@ -9227,6 +9227,43 @@ cobol85.cob:14: warning: optional period used
 AT_CLEANUP
 
 
+AT_SETUP([optional dots before PROCEDURE DIVISION])
+AT_KEYWORDS([misc missing-periods])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION
+       PROGRAM-ID. prog.
+       DATA DIVISION
+       WORKING-STORAGE SECTION
+       77 PMODE   PIC X
+       PROCEDURE DIVISION.
+          STOP RUN.
+])
+
+# The syntax error below is hard to avoid, and can be missleading.
+# The error is actually on the `77`, which is only parsed as a
+# LEVEL_NUMBER if the preceding token is a `.`.
+#
+# TODO: To avoid this, we need to unput the WORD/Literal when dealing
```

Comment:
Not sure if/what this chagnes, but after the section headers before procedure we could directly parse the error token as an alternative to `_dot` - `_dot_or_error - is the result better?

Reply:
That may work fine if we did not have the mechanics to recover using Area A. Otherwise, this work-around needs to be implemented for each kind of section. Still, it may allow the parser to pursue its work a little.

Reply:
never mind, that was just an idea.